src/backend/distributed/commands/multi_copy.c

```diff
@@ -458,12 +454,12 @@ CopyToExistingShards(CopyStmt *copyStatement, char *completionTag)
 	errorCallback.callback = CopyFromErrorCallback;
 	errorCallback.arg = (void *) copyState;
 	errorCallback.previous = error_context_stack;
-	error_context_stack = &errorCallback;
 
 	while (true)
 	{
 		ResetPerTupleExprContext(executorState);
 
+		error_context_stack = &errorCallback;
```

Comment:
Can you add a comment on why we're setting and unsetting this at every iteration?

Reply:
Hmm it seems that CopyToNewShards is doing some extra things after the copy of a row and we don't wanna show the line numbers there hence we set/unset it every iteration however here it doesn't make much sense to set/unset it every iteration because we don't do anything extra.

So I am going to close the PR, we should refactor these two methods in the future or remove CopyToNewShards at some point maybe.